### PR TITLE
[14.0][IMP] shopfloor checkout scan delivery package

### PR DESCRIPTION
**Pull request by TDu**
ref: cos-4147

### CODE REVIEW

File: shopfloor/services/checkout.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Checkout(Component):
     _description = __doc__
 
     def _response_for_select_line(
-        self, picking, message=None, need_confirm_pack_all=False
+        self, picking, message=None, need_confirm_pack_all=""
```
Comment (simahawk) [inline shopfloor/services/checkout.py:44]: IMO this should be still None when empty

**Review by simahawk — APPROVED** [review]
Minor remark

File: shopfloor/services/checkout.py [hunk 2]
```diff
@@ -620,16 +623,27 @@ def _select_lines_from_delivery_packaging(
                     packaging, carrier
                 ),
             )
+        message = None
+        need_confirm_pack_all = ""
+        lines_to_pack = selection_lines.filtered(self._filter_lines_to_pack)
+        if not lines_to_pack:
+            if self.work.menu.no_prefill_qty:
+                message = self.msg_store.no_lines_to_process_set_quantities()
+            elif not confirm_pack_all == packaging.barcode:
```
Comment (jbaudoux) [inline shopfloor/services/checkout.py]:
```suggestion
            elif confirm_pack_all != packaging.barcode:
```

File: shopfloor/services/checkout.py [hunk 3]
```diff
@@ -620,16 +623,27 @@ def _select_lines_from_delivery_packaging(
...
+                )
+            if message:
+                return self._response_for_select_line(
+                    picking,
+                    message=message,
+                    need_confirm_pack_all=need_confirm_pack_all,
+                )
         if confirm_pack_all:
-            # Select all lines and pack them all w/o passing for select_package state
             self._select_lines(selection_lines)
```
Comment (jbaudoux) [inline shopfloor/services/checkout.py]: Should be the `else` condition in the above `if not lines_to_pack:` block

File: shopfloor/services/checkout.py [hunk 4]
```diff
@@ -620,16 +623,27 @@ def _select_lines_from_delivery_packaging(
                     packaging, carrier
                 ),
             )
+        message = None
+        need_confirm_pack_all = ""
+        lines_to_pack = selection_lines.filtered(self._filter_lines_to_pack)
```
Comment (jbaudoux) [inline shopfloor/services/checkout.py]:
```suggestion
        has_lines_to_pack = any(self._filter_lines_to_pack(ml) for ml in selection_lines)
```

File: shopfloor/services/checkout.py (hunk 3 above)
Comment (TDu) [reply shopfloor/services/checkout.py]: That does not work, on a valid confirmation to pack everything there will still be nothing to pack !
I refined the if case to be more explicit.

**Review by JuMiSanAr — APPROVED** [review]

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
@TDu Can you squash the fixups in order to merge?

**Comment by TDu** [comment]
@jbaudoux squashed and rebased

**Pull request merged**